ThePensionsRegulator.Frontend/wwwroot/tpr/tpr-search.js

```diff
@@ -0,0 +1,124 @@
+    const apiUrl = searchAside.getAttribute("data-api-url");
+
+    async function initaliseAccordion() {
+        const response = await fetch(`${apiUrl}/api/faq/content/popular`, { method: 'GET', headers: { 'Content-Type': 'application/json' } });
+
+        const results = (await response.json());
+
+        if (results.length === 0) {
+            const noResultsHeading = createNoResultsFoundHeading();
+            const searchBlock = document.getElementsByClassName("tpr-search-results__input")[0];
+            searchBlock.after(noResultsHeading);
+        } else {
+            const accordionSections = [];
+            results.forEach(result => {
+                accordionSections.push(createAccordionSection(result.name, result.pageContent, result.key));
+            });
+            createAccordion(accordionSections);
+        }
+    }
+
+    function removeAccordion(id) {
+        const accordion = document.getElementById(id);
+        accordion.remove();
+    }
+
+    function createAccordion(accordionSections) {
+        const newAccordion = createElementWithClassName("div", "govuk-accordion");
+        newAccordion.setAttribute("id", "search-results-accordion");
+        newAccordion.setAttribute("data-module", "govuk-accordion");
+
+        accordionSections.forEach((accordionSection) => { newAccordion.appendChild(accordionSection); });
+
+        new Accordion(newAccordion);
+
+        const searchBlock = document.getElementsByClassName("tpr-search-results__input")[0];
+        searchBlock.after(newAccordion);
+    }
+
+    function createAccordionSection(heading, content, index) {
+        const accordionSection = createElementWithClassName("div", "govuk-accordion__section");
+
+        const accordionHeader = createElementWithClassName("div", "govuk-accordion__section-header");
+
+        const accordionHeadingElement = createElementWithClassName("h3", "govuk-accordion__section-heading"); 
+
+        const accordionHeadingElementText = document.createTextNode(heading);
+
+        const accordionHeadingButtonElement = createElementWithClassName("span", "govuk-accordion__section-button");
+        accordionHeadingButtonElement.setAttribute("id", `accordion1-heading-${index}`);
+        accordionHeadingButtonElement.appendChild(accordionHeadingElementText);
+
+        accordionHeadingElement.appendChild(accordionHeadingButtonElement);
+        accordionHeader.appendChild(accordionHeadingElement);
+
+        const accordionContent = document.createElement("div");
+        accordionContent.className = "govuk-accordion__section-content";
+        accordionContent.innerHTML = content;
+
+        accordionSection.appendChild(accordionHeader);
+        accordionSection.appendChild(accordionContent);
+
+        return accordionSection;
+    }
+
+    async function fetchContentById(contentId) {
+        const response = await fetch(`${apiUrl}/api/faq/content/${contentId}`, { method: 'GET', headers: { 'Content-Type': 'application/json' } });
```

Comment:
This component seems very tightly coupled to the API, so much that it's impossible to see the results view in the example app without the API running and configured. The example app could use a hard-coded JSON file to make the results visible, and implementing this would naturally make you implement a more decoupled solution that could support different URLs. This is how Max's autocomplete in the header search works.